### PR TITLE
introduce support for projectTemplates

### LINKED ISSUE #451: add support for /projectTemplates

### PR DESCRIPTION
**Pull request by jimirocks**
ProjectTemplateService enables to get for list of templates or particular template by URI.

Resolves #451

### CODE REVIEW

File: src/test/java/com/gooddata/projecttemplates/ProjectTemplateServiceIT.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import static net.jadler.Jadler.onRequest;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+public class ProjectTemplateServiceIT extends AbstractGoodDataIT {
+
+    private static final String TEMPLATE_URI = "/projectTemplates/ZendeskAnalytics/20";
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
```
Comment (martiner) [inline src/test/java/com/gooddata/projecttemplates/ProjectTemplateServiceIT.java]: already in parent class

File: src/test/java/com/gooddata/projecttemplates/ProjectTemplateServiceIT.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private Template template;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        template = MAPPER.readValue(readFromResource("/projecttemplates/template.json"), Template.class);
```
Comment (martiner) [inline src/test/java/com/gooddata/projecttemplates/ProjectTemplateServiceIT.java]: readObjectFromResource()

File: src/test/java/com/gooddata/projecttemplates/TemplateTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+
+public class TemplateTest {
+
+    @Test
+    public void shouldDeserialize() throws Exception {
+        final InputStream stream = getClass().getResourceAsStream("/projecttemplates/template.json");
+        final Template template = new ObjectMapper().readValue(stream, Template.class);
```
Comment (martiner) [inline src/test/java/com/gooddata/projecttemplates/TemplateTest.java]: readObjectFromResource()

File: src/main/java/com/gooddata/projecttemplates/Template.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+public class Template {
+
+    private final String uri;
+    private final String urn;
+    private final String version;
+    private final Boolean hidden;
+    private final Content content;
+    private final Meta meta;
+
+    public Template(@JsonProperty("link") String uri, @JsonProperty("urn") String urn, @JsonProperty("version") String version,
```
Comment (martiner) [inline src/main/java/com/gooddata/projecttemplates/Template.java]: does it have to be public?

File: src/test/java/com/gooddata/projecttemplates/ProjectTemplateServiceIT.java (hunk 1 above)
Comment (jimirocks) [inline src/test/java/com/gooddata/projecttemplates/ProjectTemplateServiceIT.java]: fixed

File: src/test/java/com/gooddata/projecttemplates/ProjectTemplateServiceIT.java (hunk 2 above)
Comment (jimirocks) [inline src/test/java/com/gooddata/projecttemplates/ProjectTemplateServiceIT.java]: replaced

File: src/test/java/com/gooddata/projecttemplates/TemplateTest.java (hunk 3 above)
Comment (jimirocks) [inline src/test/java/com/gooddata/projecttemplates/TemplateTest.java]: replaced

File: src/main/java/com/gooddata/projecttemplates/Template.java (hunk 4 above)
Comment (jimirocks) [inline src/main/java/com/gooddata/projecttemplates/Template.java]: changed to private

File: src/main/java/com/gooddata/projecttemplates/ProjectTemplateService.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2004-2017, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.projecttemplates;
```
Comment (liry) [inline src/main/java/com/gooddata/projecttemplates/ProjectTemplateService.java]: what about to rename it to singular "projecttemplate" the same way as others?

Comment (jimirocks) [inline src/main/java/com/gooddata/projecttemplates/ProjectTemplateService.java]: renamed

**Comment by liry** [comment]
LGTM

File: src/main/java/com/gooddata/projecttemplate/ProjectTemplateService.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            throw new GoodDataException("Unable to get template of uri=" + uri, e);
+        }
+    }
+
+    /**
+     * Get manifests of template by given uri.
+     * @param uri uri of the template
+     * @return manifests linked from project template
+     */
+    public Collection<DatasetManifest> getManifests(String uri) {
```
Comment (martiner) [inline src/main/java/com/gooddata/projecttemplate/ProjectTemplateService.java]: Template as argument

File: src/main/java/com/gooddata/projecttemplate/ProjectTemplateService.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            throw new GoodDataException("Unable to get templates", e);
+        }
+    }
+
+    /**
+     * Get project template by given uri.
+     * @param uri uri of the template
+     * @return project template
+     */
+    public Template getTemplate(String uri) {
```
Comment (martiner) [inline src/main/java/com/gooddata/projecttemplate/ProjectTemplateService.java]: getTemplateByUri

File: src/main/java/com/gooddata/projecttemplate/ProjectTemplateService.java (hunk 6 above)
Comment (jimirocks) [inline src/main/java/com/gooddata/projecttemplate/ProjectTemplateService.java]: done

File: src/main/java/com/gooddata/projecttemplate/ProjectTemplateService.java (hunk 7 above)
Comment (jimirocks) [inline src/main/java/com/gooddata/projecttemplate/ProjectTemplateService.java]: renamed

**Comment by martiner** [comment]
pls squash

**Pull request merged**